tritium-annotations/src/main/java/com/palantir/tritium/annotations/PerformanceSensitive.java

```diff
@@ -26,31 +26,34 @@
  * when modifying this element.
  */
 @Target({ElementType.TYPE, ElementType.METHOD, ElementType.MODULE})
-@Retention(RetentionPolicy.RUNTIME)
+@Retention(RetentionPolicy.SOURCE)
 public @interface PerformanceSensitive {
 
-    /**
-     * This element is sensitive to object allocations.
-     */
-    String Allocations = "Allocations";
-
-    /**
-     * This element is sensitive to cache access patterns.
-     */
-    String Cache = "Cache";
-
-    /**
-     * This element is sensitive to overall latency.
-     */
-    String Latency = "Latency";
-
-    /**
-     * This element is sensitive to overall throughput.
-     */
-    String Throughput = "Throughput";
-
     /**
      * The reasons this element is considered performance sensitive.
      */
-    String[] value();
+    Consideration[] value();
+
+    enum Consideration {
+
+        /**
+         * This element is sensitive to object allocations.
+         */
+        Allocations,
```

Comment:
Pedantic naming nits:

Perhaps allocation should be singular matching cache? As in “allocation sensitive”.

We usually use all-caps enum constants, might be worth standardizing

Reply:
good call, updated